### ISSUE TITLE
AFQMC build issue with ENABLE_CUDA using Clang and Intel 

### ISSUE THREAD

**Issue by ye-luo** [QMCPACK/qmcpack#2477]
**Describe the bug**
- [x] Clang 10 with ENABLE_CUDA turned on cannot build. Without CUDA part, the code builds.  I guess Clang 9 may also fail.
- [x] Intel 19.0 with ENABLE_CUDA turned on cannot build. Without CUDA part, the code builds.

I think code path under ENABLE_CUDA has some C++ issue.

**To Reproduce**
Steps to reproduce the behaviour:

1. Clang cmake. Using libstdc++ from gcc 7.5 and 8.2.
```
cmake -D CMAKE_C_COMPILER=mpicc -D CMAKE_CXX_COMPILER=mpicxx -D BUILD_AFQMC=1 -DENABLE_CUDA=1 -DCUDA_HOST_COMPILER=`which gcc` ..
```
clang error [CI output](https://jenkins-ci.mcs.anl.gov/me/my-views/view/QMCPACK/job/QMCPACK-CI-Clang-CUDA/label=skylake-only,type=Real-enable-CUDA/44/console)

2. Intel using libstdc++ from gcc 7.5. has issue when building unit tests. cmake
```
cmake -D CMAKE_C_COMPILER=mpicc -D CMAKE_CXX_COMPILER=mpicxx -D BUILD_AFQMC=1 -DENABLE_CUDA=1 ..
```
clang error [CI output](https://jenkins-ci.mcs.anl.gov/me/my-views/view/QMCPACK/job/QMCPACK-CI-Intel-CUDA/4/label=skylake-only,type=Complex-enable-CUDA/console)

**Expected behavior**
Build and pass unit tests.

**System:**
 - ALCF CI

**Comment by fdmalone** [comment]
clang 10 was added locally yesterday so we should be able to get to the bottom of it.

**Comment by ye-luo** [comment]
Fixed by #2480 and #2482
